tests/integration/mixer/policy/ratelimiting_test.go

```diff
@@ -112,6 +117,57 @@ func TestRateLimiting_DefaultLessThanOverride(t *testing.T) {
 		})
 }
 
+func TestRateLimiting_ExternalTraffic(t *testing.T) {
+	framework.Run(t, func(ctx framework.TestContext) {
+		destinationService := "istio-egressgateway"
+
+		bookInfoNameSpaceStr := bookinfoNs.Name()
+		config := setupConfigOrFail(t, bookinfo.CNNRedisRateLimit, bookInfoNameSpaceStr,
+			red, g, ctx)
+		g.ApplyConfigOrFail(t,
+			bookinfoNs,
+			strings.Replace(egressConfig, "<bookinfo_namespace>", bookInfoNameSpaceStr, -1))
+		defer func() {
+			deleteConfigOrFail(t, config, g, ctx)
+			g.DeleteConfigOrFail(t,
+				bookinfoNs,
+				strings.Replace(egressConfig, "<bookinfo_namespace>", bookInfoNameSpaceStr, -1))
+			util.AllowRuleSync(t)
+		}()
+		util.AllowRuleSync(t)
+
+		res, err := RunCurlTest(300, "http://edition.cnn.com/politics", bookInfoNameSpaceStr, ctx)
```

Comment:
We shouldn't be depending on external services in our tests when possible. This makes it so tests are very likely flaky, and cannot be run in some environments. is it possible to avoid this?

Reply:
It is a test for external services... Which environments won't be possible to run this?

Reply:
https://github.com/istio/istio/pull/15814, or others in network restricted environments -- I suspect users in china would not be able to run this for example. I really don't want to spend time debugging a flaky test that ends up being `cnn.com` being down or rate limiting us or something.

Reply:
Why do you need to call an external service? Can you have the same result calling an internal service through the egressgateway?